tests/end2end/e2e_sklearn.py

```diff
@@ -0,0 +1,384 @@
+                   'n_features' : n_features, 'n_classes' : n_classes}
+
+    training_args = {
+        'epochs': 5,
+        'loader_args': { 'batch_size': 1, },
+    }
+
+    tags =  ['#csv-dataset-classification']
+    rounds = 2
+
+    # search for corresponding datasets across nodes datasets
+    exp = Experiment(tags=tags,
+                     model_args=model_args,
+                     training_plan_class=SGDClassifierTrainingPlan,
+                     training_args=training_args,
+                     round_limit=rounds,
+                     aggregator=FedAverage(),
+                     save_breakpoints=True)
+    exp.run()
+    clear_experiment_data(exp)
+
+
+def test_04_sklearn_mnist_perceptron_with_declearn_optimizer():
+    """Tests SGD classifier with Declearn optimizers"""
+
+
+    model_args = {'n_features': 28*28,
+                  'n_classes' : 10,
+                  'eta0':1e-6,
+                  'random_state':1234,
+                  'alpha':0.1 }
+
+    training_args = {
+        'epochs': 3,
+        'batch_maxnum': 20,
+        'optimizer_args': {
+            "lr" : 1e-3
+        },
+        'loader_args': { 'batch_size': 4, },
+    }
+
+    tags =  ['#MNIST', '#dataset']
+    rounds = 4
+
+    # select nodes participating in this experiment
+    exp = Experiment(tags=tags,
+                     model_args=model_args,
+                     training_plan_class=SkLearnClassifierTrainingPlanDeclearn,
+                     training_args=training_args,
+                     round_limit=rounds,
+                     aggregator=FedAverage(),
+                     node_selection_strategy=None,
+                     save_breakpoints=True
+                     )
+
+    exp.run()
+    exp_folder = exp.experimentation_path()
+    del exp
+
+    loaded_exp = Experiment.load_breakpoint(os.path.join(exp_folder, 'breakpoint_0001'))
+
+    # Run starting from a breakpoints
+    loaded_exp.run_once(increase=True)
+    clear_experiment_data(loaded_exp)
+
+# Define paramters
+regressor_training_args = {
+    'epochs': 5,
+    'loader_args': { 'batch_size': 32, },
+    'test_ratio':.3,
+    'test_metric': MetricTypes.MEAN_SQUARE_ERROR,
+    'test_on_local_updates': True,
+    'test_on_global_updates': True
+}
+
+def test_05_sklearn_adni_regressor_with_declearn_optimizer():
+    """Tests declearn optimizer with sgd regressor"""
+
+
+    tags =  ['#adni']
+    rounds = 5
+
+    # select nodes participating to this experiment
+    exp = Experiment(tags=tags,
+                     model_args=regressor_model_args,
+                     training_plan_class=SGDRegressorTrainingPlanDeclearn,
+                     training_args=regressor_training_args,
+                     round_limit=rounds,
+                     aggregator=FedAverage(),
+                     node_selection_strategy=None)
+
+    exp.run()
+
+    clear_experiment_data(exp)
+
+
+
+def test_06_seklearn_adni_regressor_with_secureaggregation():
```

Comment:
Hi @ybouilla, 

Good point. I couldn't decide where to test it. I wanted to leave it here just to test if sklearn model parameters are compatible with secure aggregation since all the dataset training plan etc was already defined in this test file.